processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategySelector.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.strategy;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.metamx.common.ISE;
+import io.druid.query.groupby.GroupByQuery;
+import io.druid.query.groupby.GroupByQueryConfig;
+import io.druid.query.groupby.epinephelinae.EpiGroupByStrategy;
+
+public class GroupByStrategySelector
+{
+  public static final String CTX_KEY_STRATEGY = "groupByStrategy";
+  public static final String STRATEGY_EPINEPHELINAE = "epinephelinae";
+  public static final String STRATEGY_OLD_FAITHFUL = "oldFaithful";
+
+  private final GroupByQueryConfig config;
+  private final OldFaithfulGroupByStrategy oldFaithful;
+  private final EpiGroupByStrategy epi;
+
+  @Inject
+  public GroupByStrategySelector(
+      Supplier<GroupByQueryConfig> configSupplier,
+      OldFaithfulGroupByStrategy oldFaithful,
+      EpiGroupByStrategy epi
+  )
+  {
+    this.config = configSupplier.get();
+    this.oldFaithful = oldFaithful;
+    this.epi = epi;
+  }
+
+  public GroupByStrategy strategize(GroupByQuery query)
+  {
+    final String strategyString = query.getContextValue(CTX_KEY_STRATEGY, config.getDefaultStrategy());
+
+    switch (strategyString) {
+      case STRATEGY_EPINEPHELINAE:
+        return epi;
+
+      case STRATEGY_OLD_FAITHFUL:
+        return oldFaithful;
+
+      default:
+        throw new ISE("No such strategy[%s]", strategyString);
```

Comment:
maybe the default should be oldFaithful? This will force all queries to have a strategy